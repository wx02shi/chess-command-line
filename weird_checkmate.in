game human computer2
setup
+ R a1 + N b1 + B c1 + Q d1 + N g1 + R h1
+ P a2 + P b2 + P c2 + P d2 + P g2 + P h2
+ B b5
+ K e1
+ P f2
= black
+ R a8 + N b8 + B c8 + K e8 + R f8
+ P a7 + P d7 + P f7 + P g7 + P h7
+ P e6 + P c5
+ Q f6
done
move f2 f3
move
