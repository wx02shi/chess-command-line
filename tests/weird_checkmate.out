8 rnbqkbnr
7 pppppppp
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPPPPPP
1 RNBQKBNR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ _ _ 

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 RN_ _ _ 

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 RNB _ _ 

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 RNBQ_ _ 

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 RNBQ_ N 

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 P_ _ _ _
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PP _ _ _
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPP_ _ _
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP _ _
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP _P_
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP _PP
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP _PP
1 RNBQ_ NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP _PP
1 RNBQK NR

  abcdefgh
8  _ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 r_ _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rn _ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_ _ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_k_ _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 _ _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _ _ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_ _ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_p_ 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_pp 
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_ppp
6  _ _ _ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_ppp
6  _ _p_ _
5 _B_ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_ppp
6  _ _p_ _
5 _Bp _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_ppp
6  _ _pq _
5 _Bp _ _ 
4  _ _ _ _
3 _ _ _ _ 
2 PPPP PPP
1 RNBQK NR

  abcdefgh
Setup completed!
8 rnb_kr _
7 p _p_ppp
6  _ _pq _
5 _Bp _ _ 
4  _ _ _ _
3 _ _ _P_ 
2 PPPP _PP
1 RNBQK NR

  abcdefgh
8 rnb_kr _
7 p _p_ppp
6  _ _p_ _
5 _Bp _ _ 
4  _ _ _ q
3 _ _ _P_ 
2 PPPP _PP
1 RNBQK NR

  abcdefgh
White is in check.
Final Score:
White: 0
Black: 0
